File: 3731-find-missing-elements/3731-find-missing-elements.cpp
class Solution {
public:
    vector<int> findMissingElements(vector<int>& nums) {
        bool seen[101] = {};

        int mn = 100, mx = 0;

        for (int x : nums) {
            seen[x] = true;
            mn = min(mn, x);
            mx = max(mx, x);
        }
        vector<int> ans;
        for (int i = mn + 1; i < mx; i++) {
            if (!seen[i])
                ans.push_back(i);
        }
        return ans;
    }
};

File: 3731-find-missing-elements/3731-find-missing-elements.java
class Solution {
    public List<Integer> findMissingElements(int[] nums) {
        Set<Integer> set = new HashSet<>();

        int mn = Integer.MAX_VALUE;
        int mx = Integer.MIN_VALUE;

        for (int x : nums) {
            set.add(x);
            mn = Math.min(mn, x);
            mx = Math.max(mx, x);
        }

        List<Integer> ans = new ArrayList<>();
        for (int i = mn + 1; i < mx; i++) {
            if (!set.contains(i))
                ans.add(i);
        }
        return ans;
    }
}